packages/commons/src/auth/jwt.ts

```diff
@@ -53,22 +69,36 @@ export const verifyJwtToken = (
       jwksUri: url,
     })
   );
-  return new Promise((resolve, _reject) => {
-    jwt.verify(
-      jwtToken,
-      getKey(clients, logger),
-      {
-        audience: config.acceptedAudiences,
-      },
-      function (err, _decoded) {
-        if (err) {
-          logger.warn(`Token verification failed: ${err}`);
-          return resolve(false);
+
+  try {
+    const jwtHeader = decodeJwtTokenHeaders(jwtToken, logger);
+    if (!jwtHeader?.kid) {
+      logger.warn("Token verification failed: missing kid");
+      return Promise.reject(false);
+    }
+
+    const secret: Secret = await getKey(clients, jwtHeader.kid, logger);
```

Comment:
I would like to know if requests to the well-known are executed at every check of a JWT, I'm referring in particular to the portion of code on line 67.